functions/src/index.ts

```diff
@@ -0,0 +1,44 @@
+const functions = require("firebase-functions");
+const admin = require("firebase-admin");
+
+admin.initializeApp();
+
+exports.addAdminRole = functions.https.onCall((data: any, context: any) => {
```

Comment:
I see that these 2 functions addAdminRole and addUserRole are very similar. You could extract the lambda function from onCall and add a parameter to check the user type and use it in both cases.